code/modules/paperwork/photocopier.dm

```diff
@@ -419,6 +421,25 @@
 			. = TRUE
 	update_icon()
 
+/obj/machinery/photocopier/proc/ai_text(mob/user)
+	if(!issilicon(user))
+		return
+	if(stat & (BROKEN|NOPOWER))
+		return
+	var/text = input("Enter what you want to write:", "Write", null, null) as message
+	if(!text)
+		return
+	if(toner < 1 || !user)
+		return
+	playsound(loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, TRUE)
+	var/obj/item/paper/p = new /obj/item/paper(loc)
+	text = p.parsepencode(text, null, user)
+	p.info = text
+	p.populatefields()
+	toner -= 1
+	use_power(active_power_consumption)
+	sleep(PHOTOCOPIER_DELAY)
```

Comment:
Having a sleep here isnt the best idea.

Consider tracking the `world.time` and making sure `PHOTOCOPIER_DELAY` has passed before allowing it again.

Reply:
Agreed, but use the tg cooldowns system. Stuff like COOLDOWN_DECLARE, COOLDOWN_START, etc